src/material-experimental/mdc-checkbox/checkbox.ts

```diff
@@ -0,0 +1,296 @@
+    isAttachedToDOM: () => !!this._checkbox.nativeElement.parentNode,
+    isChecked: () => this.checked,
+    isIndeterminate: () => this.indeterminate,
+    removeNativeControlAttr:
+        (attr) => {
+          if (!this._attrBlacklist.has(attr)) {
+            this._nativeCheckbox.nativeElement.removeAttribute(attr);
+          }
+        },
+    setNativeControlAttr:
+        (attr, value) => {
+          if (!this._attrBlacklist.has(attr)) {
+            this._nativeCheckbox.nativeElement.setAttribute(attr, value);
+          }
+        },
+    setNativeControlDisabled: (disabled) => this.disabled = disabled,
+  };
+
+  constructor(
+      private _cdr: ChangeDetectorRef, private _platform: Platform, private _ngZone: NgZone,
+      @Attribute('tabindex') tabIndex: string,
+      @Optional() @Inject(MAT_CHECKBOX_CLICK_ACTION) private _clickAction: MatCheckboxClickAction,
+      @Optional() @Inject(ANIMATION_MODULE_TYPE) public _animationMode?: string) {
+    this.tabIndex = parseInt(tabIndex) || 0;
+    this._checkboxFoundation = new MDCCheckboxFoundation(this._checkboxAdapter);
+    // Note: We don't need to set up the MDCCheckboxFoundation. Its only purpose is to manage the
+    // ripple, which we do ourselves instead.
+  }
+
+  ngAfterViewInit() {
+    this._rippleTarget = {
+      rippleConfig: {
+        radius: 20,
+        centered: true,
+        animation: {
+          // TODO(mmalerba): Use the MDC constants once they are exported separately from the
+          // foundation. Grabbing them off the foundation prevents the foundation class from being
+          // tree-shaken.
+          enterDuration: 225 /* MDCRippleFoundation.numbers.DEACTIVATION_TIMEOUT_MS */,
+          exitDuration: 150 /* MDCRippleFoundation.numbers.FG_DEACTIVATION_MS */,
+        },
+      },
+      rippleDisabled: true
+    };
+    this._rippleRenderer =
+        new RippleRenderer(this._rippleTarget, this._ngZone, this._checkbox, this._platform);
+
+    this._checkboxFoundation.init();
+  }
+
+  ngOnDestroy() {
+    this._checkboxFoundation.destroy();
+  }
+
+  registerOnChange(fn: (checked: boolean) => void) {
+    this._cvaOnChange = fn;
+  }
+
+  registerOnTouched(fn: () => void) {
+    this._cvaOnTouch = fn;
+  }
+
+  setDisabledState(isDisabled: boolean) {
+    this.disabled = isDisabled;
+  }
+
+  writeValue(value: any) {
+    this.checked = !!value;
+    this._cdr.markForCheck();
+  }
+
+  focus() {
+    this._nativeCheckbox.nativeElement.focus();
+  }
+
+  toggle() {
+    this.checked = !this.checked;
+  }
+
+  _activateRipple() {
+    if (!this.disabled && !this.disableRipple && this._animationMode != 'NoopAnimations') {
+      this._rippleRenderer.fadeInRipple(0, 0, this._rippleTarget.rippleConfig);
+    }
+  }
+
+  _onBlur() {
+    Promise.resolve().then(() => {
+      this._cvaOnTouch();
+      this._cdr.markForCheck();
+    });
+  }
+
+  // Note: we must listen to the `click` event rather than the `change` event because IE & Edge do
```

Comment:
is this an angular specific eventing system issue? 

Seems to work on the [vanilla components](https://material-components.github.io/material-components-web-catalog/#/component/checkbox)